logical_switch_port_test.go

```diff
@@ -147,13 +158,119 @@ func TestLogicalSwitchPortAPI(t *testing.T) {
 	assert.Equal(t, ok, true)
 	assert.Equal(t, extIdMac, PORT_TEST_EXT_ID_MAC)
 
+	extIdIP, ok := extIdsRet[PORT_TEST_EXT_ID_IP_KEY]
+	assert.Equal(t, ok, true)
+	assert.Equal(t, extIdIP, PORT_TEST_EXT_ID_IP)
+
+	t.Logf("Validated the external ids are set correctly")
+	// update one of the existing keys, remove one key
+	// and make sure it's a complete update.
+	t.Logf("Validating that keys get clobbered by Set API correctly")
+	extIds[PORT_TEST_EXT_ID_MAC_KEY] = PORT_TEST_EXT_ID_MAC_2
+	delete(extIds, PORT_TEST_EXT_ID_IP_KEY)
```

Comment:
Hi @abhat : Sorry may I couldn't point correctly. What I meant is if you set extids/options for a k:v , existing k:v should not be impacted.
If you want to delete the extIds/Options, do it with delAPI not with just dict delete(). e.g. if there are two extIDs: a:b , c:d, if you set c=f, a:b should be present too. That is why I gave the e.g. of nbctl :
e.g `current extIDs: external_ids        : {aa=yyy, abc="456", jjj="777", xxx="123", zzz="123"}`

`run ovn-nbctl  set logical_switch_port sw0-port1 external-ids:zzz=125`

`final output: external_ids        : {aa=yyy, abc="456", jjj="777", xxx="123", zzz="125"}`
You can also refer to dhcp_options.go where it implements Add,set,del for options for ref. In this PR, we don't even have add/del api support for extid yet and only set. Hence, you keep setting multiple Keys but other keys should be intact.
Also for Setting options/ext ID, don't do get first. You can set it with map. e.g. refer to dhcp_options_test.go for ref. Correct me if I am wrong.

Reply:
I think you are missing the semantic of `SetOptions` or `SetExternalIds`. I thought we agreed that a Set API should do a clobbering of existing keys. And an `AddOption`/`DelOption` API should do the addition/deletion of individual keys.

Reply:
The GetOptions or GetExternalIds in the unit-test is just to make sure that the default value for the field in the row is not nil but it's of type OvsMap.

Reply:
Here is what @hzhou8 had to say on the issue: https://github.com/ebay/go-ovn/pull/84#discussion_r417015003

Reply:
Thanks @abhat . Yup, I align with the discussion that @hzhou8 pointed. But after cross checking ovn-nbctl commands, it conflicts with our consent of full update on the column where set on a key just updates value of existing key no mutation to other keys. If we are still ok with current approach,  it's lgtm from my side. @hzhou8 : Feel free to take a final look and merge if no more suggestions.

Reply:
Yes, my first comment regarding this was from ovn-nbctl alike command perspective. However, I realized that in the API names convention, set usually means full update. It is better to keep that convention aligned at this lib API level. We can add UpdateOptionsAddKey, UpdateOptionsDelKey, UpdateOptionsUpdateKey, etc. later for the operations where full update is not desired, but again since we have a plan for auto-generating APIs, let's don't waste too much time for operations not really needed. For now I think the API of current patch is good enough.

Reply:
Sounds good. Thanks for confirming.